hack/verify-structured-logging.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+export KUBE_ROOT
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+migrated_files="${KUBE_ROOT}/hack/.structured_logging"
+migrated_packages=$(sed "s| | -e |g" "${migrated_files}")
+
+# Install logcheck
+pushd "${KUBE_ROOT}/hack/tools" >/dev/null
+  GO111MODULE=on go install k8s.io/klog/hack/tools/logcheck
+popd >/dev/null
+
+for package in $migrated_packages
+do
+	echo "verifying structured logging for $package"
+	find "$KUBE_ROOT/$package" -type d -print0 | xargs -0 logcheck || continue
```

Comment:
I don't think we need the find call here at all actually, in the future we're going to need to use k8s.io/kubernetes/$pkg or ./$pkg anyhow.

we can just do `logcheck $pkg` where $pkg is `./pkg/foo`

Reply:
currently linter is not working recursively and it expects complete path.

Reply:
oh I see, the file is only listing top level directories to enforce?

I think we should have a comment somewhere explaining that intent, usually we list individual packages that are excluded, or individual files.

Reply:
yes, right it is only listing top level directories. Sure will add a comment here.

Reply:
Updated .structured_logging to add all directories, not going against pattern here :). Previous approach was confusing.